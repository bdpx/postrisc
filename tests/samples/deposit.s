.text
    write	"test bit-field insert (deposit)"
    alloc	96
    ldi.l	%r30, 0xaaaaaaaaaaaaaaaa
    ldi.l	%r40, 0xeeeeeeeeeeeeeeee
    dep		%r20, %r30, %r40, 48, 24
    write	"dep: %x64(r20)"
    dep		%r20, %r40, %r30, 48, 24
    write	"dep: %x64(r20)"

    write	"test vector deposit (dep16)"
    nor		%r3, %r4, %r4
    dep.q	%r5, %r3, %r4, 100
    write	"dep16: %x128(r5)"
    write	"end deposit test"
.end
