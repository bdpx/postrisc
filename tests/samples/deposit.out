test bit-field insert (deposit)
dep: eeeeaaaaaaaaaaee
dep: aaaaeeeeeeeeeeaa
test vector deposit (dep16)
dep16: 0000000fffffffff_ffffffffffffffff
end deposit test
